In [2]:

pip install pandas numpy scipy statsmodels seaborn matplotlib scikit-learn pingouin pyreadstat


  Using cached joblib-1.5.2-py3-none-any.whl.metadata (5.6 kB)
  Using cached threadpoolctl-3.6.0-py3-none-any.whl.metadata (13 kB)
   ---------------------------------------- 0.0/8.7 MB ? eta -:--:--
   ---------------------------------------  8.7/8.7 MB 56.6 MB/s eta 0:00:01
   ---------------------------------------- 8.7/8.7 MB 38.3 MB/s eta 0:00:00
Using cached joblib-1.5.2-py3-none-any.whl (308 kB)
Using cached threadpoolctl-3.6.0-py3-none-any.whl (18 kB)
   ---------------------------------------- 0.0/1.4 MB ? eta -:--:--
   ---------------------------------------- 1.4/1.4 MB 27.3 MB/s eta 0:00:00

   ----- ---------------------------------- 1/8 [tabulate]
   ---------- ----------------------------- 2/8 [joblib]
   ---------- ----------------------------- 2/8 [joblib]
   ---------- ----------------------------- 2/8 [joblib]
   ---------- ----------------------------- 2/8 [joblib]
   ---------- ----------------------------- 2/8 [joblib]
   ---------- ----------------------------- 


[notice] A new release of pip is available: 25.1.1 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


In [4]:
import pyreadstat

df, meta = pyreadstat.read_sav("Obesity.sav")
print(df.columns.tolist())
df.info()
df.head()
df.describe()



['Gender', 'Age', 'Race', 'Marital_status', 'Education', 'Occupation', 'Household_income', 'Weight', 'Height', 'WC', 'WC_coded', 'Systolic_BP', 'Diastolic_BP', 'Confectionary_scale', 'Nutrition_Confectionary', 'Fruits_scale', 'Nutrition_Fruits', 'Vege_scale', 'Nutrition_Veggies', 'BMI', 'BMI_WHO', 'BMI_CPG', 'BMI_2cat', 'BMI_2cat_WHO', 'BMI_2cat_overweight', 'BMI_2cat_overweightCPG']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 26 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Gender                   100 non-null    float64
 1   Age                      100 non-null    float64
 2   Race                     100 non-null    float64
 3   Marital_status           100 non-null    float64
 4   Education                100 non-null    float64
 5   Occupation               100 non-null    float64
 6   Household_income         100 non-null    float64
 7   Weight            

Gender         Age        Race  Marital_status   Education  \
count  100.000000  100.000000  100.000000      100.000000  100.000000   
mean     1.700000   50.350000    1.890000        1.970000    2.840000   
std      0.460566   15.907799    0.962898        0.744746    0.861289   
min      1.000000   18.000000    1.000000        1.000000    1.000000   
25%      1.000000   40.750000    1.000000        2.000000    2.000000   
50%      2.000000   53.000000    1.000000        2.000000    3.000000   
75%      2.000000   62.250000    3.000000        2.000000    3.000000   
max      2.000000   80.000000    3.000000        4.000000    5.000000   

       Occupation  Household_income      Weight      Height          WC  ...  \
count  100.000000        100.000000  100.000000  100.000000  100.000000  ...   
mean     3.950000       1898.000000   67.285400  156.310000   92.910000  ...   
std      1.971386       1272.413598   14.586868   10.138132   13.929908  ...   
min      1.000000        150.000000   41.000000  130.000000   43.000000  ...   
25%      2.000000       1000.000000   57.015000  149.000000   84.000000  ...   
50%      3.500000       1750.000000   64.385000  156.000000   93.000000  ...   
75%      6.000000       2500.000000   76.725000  162.000000  100.000000  ...   
max      7.000000       6500.000000  112.000000  183.000000  131.000000  ...   

       Nutrition_Fruits  Vege_scale  Nutrition_Veggies         BMI  BMI_WHO  \
count        100.000000  100.000000         100.000000  100.000000    100.0   
mean           3.420000    5.450000           5.450000   27.535253      3.1   
std            2.567001    2.262452           2.262452    5.306404      1.0   
min            0.000000    0.000000           0.000000   17.239048      1.0   
25%            1.000000    4.000000           4.000000   23.643560      2.0   
50%            3.000000    7.000000           7.000000   27.427771      3.0   
75%            7.000000    7.000000           7.000000   30.823225      4.0   
max            7.000000    7.000000           7.000000   40.750386      6.0   

          BMI_CPG    BMI_2cat  BMI_2cat_WHO  BMI_2cat_overweight  \
count  100.000000  100.000000    100.000000           100.000000   
mean     3.470000    1.490000      1.310000             1.280000   
std      1.158674    0.502418      0.464823             0.451261   
min      1.000000    1.000000      1.000000             1.000000   
25%      3.000000    1.000000      1.000000             1.000000   
50%      3.000000    1.000000      1.000000             1.000000   
75%      4.000000    2.000000      2.000000             2.000000   
max      6.000000    2.000000      2.000000             2.000000   

       BMI_2cat_overweightCPG  
count              100.000000  
mean                 1.280000  
std                  0.451261  
min                  1.000000  
25%                  1.000000  
50%                  1.000000  
75%                  2.000000  
max                  2.000000  

[8 rows x 26 columns]

In [6]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 26 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Gender                   100 non-null    float64
 1   Age                      100 non-null    float64
 2   Race                     100 non-null    float64
 3   Marital_status           100 non-null    float64
 4   Education                100 non-null    float64
 5   Occupation               100 non-null    float64
 6   Household_income         100 non-null    float64
 7   Weight                   100 non-null    float64
 8   Height                   100 non-null    float64
 9   WC                       100 non-null    float64
 10  WC_coded                 100 non-null    float64
 11  Systolic_BP              100 non-null    float64
 12  Diastolic_BP             100 non-null    float64
 13  Confectionary_scale      100 non-null    float64
 14  Nutrition_Confectionary  10

In [5]:
df['variable'].value_counts()
df['variable'].value_counts(normalize=True)*100



KeyError: 'variable'


#  **Statistical Analysis Plan (SAP)**

## **1. Study Question and Aims**

**Primary question:**
What is the association between dietary patterns (and diet quality scores) and the prevalence of hypertension?

**Aims:**

1. Describe demographic, anthropometric, and dietary characteristics of the sample.
2. Compare the prevalence of hypertension across dietary groups and diet-quality levels.
3. Determine whether dietary patterns or diet-quality scores predict hypertension after adjusting for confounders.
4. Conduct subgroup and sensitivity analyses to test robustness.


## **2. Overview of Analytic Approach**

* Create a binary **Hypertension** variable (SBP ≥130 or DBP ≥80).
* Perform descriptive statistics (means/SD, medians/IQR, frequencies).
* Conduct univariate tests: chi-square, t-tests, Mann–Whitney U.
* Build logistic regression models (progressive adjustment).
* Perform diagnostics: VIF, ROC/AUC, calibration.
* Conduct subgroup + interaction tests.
* Produce tables and visualizations.


## **3. Data Preparation & Initial Exploration**

### **3.1 Variable Definitions**

* **Outcome:**

  * `Hypertension` = 1 if Systolic_BP ≥130 or Diastolic_BP ≥80; else 0.
* **Primary exposures:**

  * Dietary pattern variable, and/or dietary scales (Fruits, Vegetables, Confectionary).
* **Covariates:**

  * Age, Gender, Race, BMI, Education, Income, Occupation, WC, etc.

### **3.2 Missing Data Plan**

* Summarize missingness with `df.isnull().sum()`.
* If **<5%** missing → complete-case analysis.
* If **≥5%** and missingness appears at-random → consider multiple imputation.
* Imputation model should include all variables used in analysis.

### **3.3 Outliers & Distributions**

* Use histograms, boxplots, Q-Q plots for continuous data.
* Identify outliers via:

  * z-scores (|z| > 3)
  * IQR rule (Q1 − 1.5×IQR, Q3 + 1.5×IQR)
* Consider transformations or non-parametric tests for skewed data.
* Conduct sensitivity analyses with outliers removed.

### **3.4 Multicollinearity Assessment**

* Compute VIF for continuous predictors.
* VIF > 5 indicates potentially problematic multicollinearity.
* If multiple highly correlated dietary scales exist, consider:

  * Using single scales,
  * Creating composite indices, or
  * Running PCA for dietary patterns.


## **4. Univariate Analyses**

### **4.1 Descriptive Statistics**

* **Table 1:** Sample characteristics

  * Continuous: mean ± SD or median [IQR]
  * Categorical: n (%)
* Include overall hypertension prevalence.

### **4.2 Univariate Comparisons**

* **Categorical × categorical:** Chi-square (or Fisher’s exact).
* **Continuous × binary:** t-test (if normal) or Mann–Whitney U (if non-normal).
* Report p-values and effect sizes.


## **5. Multivariable Modeling Strategy**

### **5.1 Progressive Logistic Regression Models**

Report ORs and 95% CIs.

**Model 0 (Unadjusted):**
`Hypertension ~ Dietary_variable`

**Model 1 (Demographics):**
`Hypertension ~ Dietary_variable + Age + Gender`

**Model 2 (SES Adjustment):**
`Hypertension ~ Dietary_variable + Age + Gender + Race + Education + Household_income`

**Model 3 (Full Model):**
`Hypertension ~ Dietary_variable + Age + Gender + Race + Education + Household_income + BMI + WC`

### **5.2 Exposure Parameterizations**

* Diet variables as continuous (OR per SD increase).
* Categorical versions (tertiles/quartiles).
* If using PCA: include PC1, PC2 as exposures.


## **6. Model Diagnostics & Performance**

### **6.1 Assumptions & Fit**

* Check linearity of continuous predictors with log-odds.
* Calculate VIF for multicollinearity.
* Detect influential observations (Cook’s distance, leverage).
* Rerun models excluding influential points if needed.

### **6.2 Model Fit & Discrimination**

* Hosmer–Lemeshow goodness-of-fit test.
* ROC curve and AUC for discrimination.
* Calibration plots (predicted vs observed probabilities).


## **7. Subgroup & Interaction Analyses**

### **7.1 Subgroups**

Run stratified logistic models for:

* Sex (Male/Female)
* Age groups (<50 vs ≥50)
* BMI categories (Normal, Overweight/Obese)

### **7.2 Interaction Terms**

Include:

* `Dietary_variable * Gender`
* `Dietary_variable * Age_group`
  Report interaction p-values and stratified ORs.


## **8. Sensitivity Analyses**

* Use alternative hypertension thresholds (e.g., ≥140/90).
* Exclude outliers and re-run models.
* Compare complete-case vs imputed models if applicable.
* Run models with individual diet variables vs grouped diet variables.


## **9. Multiple Testing Considerations**

* Main hypotheses remain primary analyses.
* Exploratory analyses may use FDR correction if needed.
* Emphasize effect sizes and confidence intervals over p-values alone.


## **10. Planned Tables and Figures**

### Tables

1. **Table 1:** Descriptive characteristics.
2. **Table 2:** Hypertension prevalence by dietary groups.
3. **Table 3:** Logistic regression results for Models 0–3.
4. Supplemental: VIF table, sensitivity analysis tables.

### Figures

1. Figure 1: Hypertension prevalence by dietary pattern (bar chart).
2. Figure 2: Scatterplot of diet score vs Systolic BP with regression line.
3. Supplemental: correlation heatmap, ROC curve, calibration plot.


## **11. Statistical Software**

Analyses will be performed using:

* `pandas`, `numpy`
* `matplotlib`, `seaborn`
* `scipy.stats`
* `statsmodels.api` & `statsmodels.formula.api`
* `sklearn.metrics`


